let res,models,NotFound;_14a‍.x([["saveUser",()=>saveUser],["getAllUsers",()=>getAllUsers],["update",()=>update],["deleteById",()=>deleteById]]);_14a‍.w("express/lib/response",[["default",["res"],function(v){res=v}]]);_14a‍.w("../models",[["default",["models"],function(v){models=v}]]);_14a‍.w("../utils/errors",[["NotFound",["NotFound"],function(v){NotFound=v}]]);



       const saveUser = async (user) => {
    const model = new models.User(user);
    const savedUser = await model.save();
    return savedUser;
};

       const getAllUsers = async () => {
    const User = models.User;
    const users = await User.find();
    return users;
}
       const update = async (user) => {
    const id = user._id;
    const User = models.User;
    let model = await User.findById(id);
    if (model) {
        model.username = user.username;
        model.save();
        return model;
    }
    throw new NotFound("update not happended in id " + user._id);
}

       const deleteById = async (id) => {
    const User = models.User;
    _14a‍.g.console.log("service" + id);
    if (id.length == 24) {
        let model = await User.findById(id);
        if (model) {
            let result = await User.deleteOne({ _id: id });
            return result;
        }
        throw new NotFound('User not found by the id ' + id);
    }
    else {

        throw new NotFound('User not found by the id ' + id);

    }



}